build-system/pr-check.js

```diff
@@ -257,27 +259,25 @@ const command = {
 
 function runAllCommands() {
   // Run different sets of independent tasks in parallel to reduce build time.
-  if (process.env.BUILD_SHARD == "pre_build_checks") {
+  if (process.env.BUILD_SHARD == "pre_build_checks_and_unit_tests") {
     command.testBuildSystem();
-    command.runPreBuildChecks();
+    command.cleanBuild();
+    command.buildRuntime();
+    command.runLintChecks();
     command.runDepAndTypeChecks();
-    // Skip testDocumentLinks() during push builds.
+    command.runUnitTests();
+    // command.testDocumentLinks() is skipped during push builds.
     command.buildValidatorWebUI();
     command.buildValidator();
   }
   if (process.env.BUILD_SHARD == "integration_tests") {
+    command.cleanBuild();
     command.buildRuntime();
-    command.presubmit();  // Must be run after the runtime is built.
+    command.serveRuntime();
+    command.runPresubmitTests();  // Needs runtime to be built and served.
```

Comment:
This means we'd actually be running "presubmit" post-submit. 😄  

Why does presubmit require dist? Can we run it across the `src/` dir at least?